### PR TITLE
[release/5.0] Resolve Component Governance bugs

### PR DESCRIPTION
**Pull request by wtgodbe**
Part of https://github.com/dotnet/aspnetcore/issues/37889. Resolves the remaining High/Critical 5.0 CompGov bug(s)

### CODE REVIEW

File: src/Razor/test/testassets/nuget.config [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
+  	<clear />
```
Comment (dougbu) [inline src/Razor/test/testassets/nuget.config:4]: This means the feeds from our root NuGet.config file aren't visible to the test asset projects. Entirely expected for the build to fail w/o those feeds

**Comment by Pilchie** [comment]
:eyes:

File: src/Razor/test/testassets/ClassLibraryMvc21/ClassLibraryMvc21.csproj [hunk 2]
```diff
@@ -14,5 +14,6 @@
       <!-- Avoid exporting types from real MVC that will conflict with the test shim -->
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="System.Text.Encodings.Web" Version="4.5.1" />
```
Comment (dougbu) [inline src/Razor/test/testassets/ClassLibraryMvc21/ClassLibraryMvc21.csproj:17]: To confirm: This and SimpleMvc21NetFx.csproj` don't import ResolveReferences.targets❔ If that's the case, this is fine.

File: .azure/pipelines/jobs/default-build.yml [hunk 3]
```diff
@@ -155,9 +155,9 @@ jobs:
         displayName: Start background dump collection
     - ${{ if eq(parameters.installNodeJs, 'true') }}:
       - task: NodeTool@0
-        displayName: Install Node 10.x
+        displayName: Install Node 14.x
         inputs:
-          versionSpec: 10.x
+          versionSpec: 14.x
```
Comment (dougbu) [inline .azure/pipelines/jobs/default-build.yml:160]: Should we jump all the way forward to 16.x and catch up w/ 'main'❔ Thoughts @javiercn @BrennanConroy&hellip;

**Review by dougbu — COMMENTED** [review]
Looks fine but please don't merge w/o signoff from someone deeper in our `node`, `npm` / `yarn` world.

File: src/Razor/test/testassets/ClassLibraryMvc21/ClassLibraryMvc21.csproj (hunk 2 above)
Comment (wtgodbe) [reply src/Razor/test/testassets/ClassLibraryMvc21/ClassLibraryMvc21.csproj:17]: Correct

File: .azure/pipelines/jobs/default-build.yml (hunk 3 above)
Comment (wtgodbe) [reply .azure/pipelines/jobs/default-build.yml:160]: I did this to match 6.0: https://github.com/dotnet/aspnetcore/blob/d5d2e4da9e1371533c27d073a09012ff96aeeffb/.azure/pipelines/jobs/default-build.yml#L169-L173

**Review by dougbu — APPROVED** [review]

**Pull request merged**